src/back-end/index.ts

```diff
@@ -308,5 +314,21 @@ export async function stopServer() {
     process.exit(1);
   }
 
-  await server.close();
+  try {
+    await new Promise<void>((resolve, reject) => {
+      server.close((error) => {
+        if (error) {
+          reject(error);
+        } else {
+          connectToDatabase()
+            .destroy()
+            .then(() => resolve())
+            .catch(reject);
```

Comment:
Is this required because of the how Knex handles the connections internally? It seems like it would create a new closure (with a new instance of `_connection`) that would immediately be destroyed, but I'm guessing it's being done to force to Knex to clean up?

Reply:
Yeah, I was a little worried about that too, but when I inspected the running program they connections created _seem_ to reference the same object, i.e. when compared using ===.